### PR TITLE
Read all the usage counts when we retrieve the variables

### PR DESCRIPTION
**Pull request by Br3nda**

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -325,6 +334,7 @@ DEPENDENCIES
   bootstrap (~> 4.2.1)
   byebug
   capybara (>= 2.15)
+  capybara-screenshot
```
Comment (mischa-s) [inline Gemfile.lock]: :computer: excited for this to be introduced once it is being used, for now though maybe we don't include?

File: spec/controllers/variables_controller_spec.rb [hunk 2]
```diff
@@ -3,4 +3,21 @@
...
+  describe '#index' do
+    context 'two variables that refer to each other' do
+      let!(:parent) { FactoryBot.create :variable, name: 'likes_to_eat_chocolate', variables: [child] }
+      let!(:child) { FactoryBot.create :variable, name: 'likes_to_eat' }
+      before { get :index, params: {} }
+      it { expect(assigns(:variables)).to eq [child, parent] }
+      it { expect(assigns(:link_from_counts)).to eq ({ child.id => 1 }) }
+      it { expect(assigns(:link_to_counts)).to eq ({ parent.id => 1 }) }
+    end
+
```
Comment (mischa-s) [inline spec/controllers/variables_controller_spec.rb]: Do any other variables exist in scope at this time to make sure it is testing for links when multiple exist?

File: spec/controllers/variables_controller_spec.rb [hunk 3]
```diff
@@ -3,4 +3,21 @@
...
+    context 'one orphaned variable' do
+      let!(:variable) { FactoryBot.create :variable }
+      before { get :index, params: {} }
+      it { expect(assigns(:link_from_counts)).to eq ({}) }
+      it { expect(assigns(:link_to_counts)).to eq ({}) }
```
Comment (mischa-s) [inline spec/controllers/variables_controller_spec.rb]: Do more variables exist in scope at this point for it to falsely link to?

Comment (Br3nda) [reply spec/controllers/variables_controller_spec.rb]: No, the spec is nicely isolated

Comment (Br3nda) [reply spec/controllers/variables_controller_spec.rb]: I've added a more complicated scenarios in controller specs

File: app/controllers/variables_controller.rb [hunk 4]
```diff
@@ -4,14 +4,28 @@ class VariablesController < ApplicationController
   before_action :set_variable, only: %i[show edit update destroy]
   def index
     @search = VariablesSearchService.new(params)
-    @variables = @search.result.distinct(false).includes(:variables, :value_type).order(:name)
+
+    @variables = @search.result
+                        .distinct(false)
+                        .includes(:variables, :value_type, :entity)
+                        .order(:name)
+
```
Comment (mischa-s) [inline app/controllers/variables_controller.rb]: :book: nice and readable

File: spec/controllers/variables_controller_spec.rb [hunk 5]
```diff
@@ -3,4 +3,47 @@
 require 'rails_helper'
 
 RSpec.describe VariablesController, type: :controller do
+  describe '#index' do
```
Comment (mischa-s) [inline spec/controllers/variables_controller_spec.rb]: :star2:

**Pull request merged**